content-src/lib/embedly-middleware.js

```diff
@@ -0,0 +1,53 @@
+const am = require("actions/action-manager");
+const isDevServerRunning = !!__CONFIG__.API_KEY;
+
+function buildQuery(items) {
+  return items
+    .map(item => item.url)
+    .map(encodeURIComponent)
+    .map(url => "urls=" + url)
+    .join("&");
+}
+
+const actionsToSupplement = new Set([
+  "TOP_FRECENT_SITES_RESPONSE",
+  "RECENT_BOOKMARKS_RESPONSE",
+  "RECENT_HISTORY_RESPONSE"
+].map(type => am.type(type)));
+
+module.exports = () => next => action => {
+  // We don't want to add extra data if the response is an error
+  if (action.error) {
```

Comment:
i think i'd prefer keeping the conditions separate, so they are easier to remove separately and comment